blocks/pdf-viewer/pdf-viewer.js

```diff
@@ -0,0 +1,141 @@
+          adobeDCView.previewFile({
+            content: { location: { url: docUrl } },
+            metaData: { fileName: docFilename },
+          }, {
+            embedMode,
+            showAnnotationTools,
+            showPrintPDF,
+            showDownloadPDF,
+            defaultViewMode,
+            enableFormFilling,
+            showBookmarks,
+            showThumbnails,
+            showZoomControl,
+            showLeftHandPanel,
+          });
+
+          // Add data-hj-allow-iframe attribute to the generated iframe
+          const iframe = document.querySelector(`#${divId} iframe`);
+          if (iframe) {
+            iframe.setAttribute('data-hj-allow-iframe', '');
+          }
+        }
+      });
+    }
+  }
+};
+
+export default async function decorate(block) {
+  const blockConfig = readBlockConfig(block);
+  const docUrl = blockConfig['document-link'];
+  const embedMode = blockConfig['embed-mode'];
+  const showAnnotationTools = (blockConfig['show-annotation-tools'] === 'true');
+  const showPrintPDF = (blockConfig['show-print-pdf'] === 'true');
+  const showDownloadPDF = (blockConfig['show-download-pdf'] === 'true');
+  const defaultViewMode = blockConfig['default-view-mode'];
+  const enableFormFilling = (blockConfig['enable-form-filling'] === 'true');
+  const showBookmarks = (blockConfig['show-bookmarks'] === 'true');
+  const showThumbnails = (blockConfig['show-thumbnails'] === 'true');
+  const showZoomControl = (blockConfig['show-zoom-control'] === 'true');
+  const showLeftHandPanel = (blockConfig['show-left-hand-panel'] === 'true');
+
+  if (docUrl) {
+    const randomUUID = window.crypto.randomUUID();
+    const divId = `adobe-dc-view-${randomUUID}`;
+    const docDiv = createTag('div', { id: divId });
+
+    // Add embed mode class
+    if (embedMode) {
+      const embedModeCleanedUp = embedMode.replace(/[_\s]/g, '-').toLowerCase();
+      const concatenatedClass = `embed-mode--${embedModeCleanedUp}`;
+      block.classList.add(concatenatedClass);
+    } else {
+      block.classList.add('embed-mode--full-window');
+    }
+    block.textContent = '';
+    block.append(docDiv);
+
+    // Get document asset type
+    const assetType = getMetadata('assettype');
+    const isWhitepaper = assetType.toLowerCase() === 'whitepaper';
+    const isSolutionBrief = assetType.toLowerCase() === 'solution brief';
+
+    // If the assetType is 'Whitepaper' or 'Solution Brief'
+    if (isWhitepaper || isSolutionBrief) {
+      block.classList.add('pdf-viewer-orientation--portrait');
+    } else {
+      block.classList.add('pdf-viewer-orientation--landscape');
+    }
+
+    window.setTimeout(() => embedPDFViewer(
+      divId,
+      docUrl,
+      embedMode,
+      showAnnotationTools,
+      showPrintPDF,
+      showDownloadPDF,
+      defaultViewMode,
+      enableFormFilling,
+      showBookmarks,
+      showThumbnails,
+      showZoomControl,
+      showLeftHandPanel,
+    ), 3000);
```

Comment:
let's leave it at 3000 and I can force-merge it. That way we can see how it behaves in the main branch .hlx.page.